pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -272,6 +330,8 @@ func (r *Reconciler) getOrCreateClusterResourceSnapshot(ctx context.Context, crp
 			},
 			Annotations: map[string]string{
 				fleetv1beta1.ResourceGroupHashAnnotation: resourceHash,
+				// TODO need to updated once we support multiple snapshots
```

Comment:
looks like the PR assumes more than one resourceSnapshot in an index but the create part only handles one?